orderer/common/multichannel/registrar.go

```diff
@@ -401,8 +401,20 @@ func (r *Registrar) ChannelList() types.ChannelList {
 }
 
 func (r *Registrar) ChannelInfo(channelID string) (types.ChannelInfo, error) {
-	//TODO
-	return types.ChannelInfo{}, errors.New("Not implemented yet")
+	r.lock.RLock()
+	defer r.lock.RUnlock()
+
+	info := types.ChannelInfo{}
+	cs, ok := r.chains[channelID]
+	if !ok {
+		return info, types.ErrChannelNotExist
+	}
+
+	info.Name = channelID
```

Comment:
can't you initialize them in a single directive?

Reply:
am not sure what you mean... like so:
```go
info.Name, info.Height = channelID, cs.Height()
```

versus

```go
info.Name = channelID
info.Height = cs.Height()
```

Reply:
no, like: info := &InfoObject{ .... }

Reply:
there is already an empty info object above that I return in case of error, here I just fill it up... not sure what else I can do. This saves only a few characters anyway.